lib/govuk_web_banners/test_validators/recruitment_banner.rb

```diff
@@ -0,0 +1,54 @@
+require "active_support/core_ext/integer/time"
+
+module GovukWebBanners
+  module TestValidators
+    class RecruitmentBanner
+      attr_reader :errors
+
+      def initialize(banners)
+        @errors = {}
+
+        validate(banners)
+      end
+
+      def valid?
+        @errors.keys.none?
+      end
+
+    private
+
+      def add_error(banner, error)
+        @errors[safe_name(banner)] ||= []
+        @errors[safe_name(banner)] << error
+      end
+
+      def safe_name(banner)
+        banner.name || "unnamed banner"
+      end
+
+      def validate(banners)
+        banners.each do |banner|
+          add_error(banner, "is missing a suggestion_text") unless banner.suggestion_text.present?
+          add_error(banner, "is missing a suggestion_link_text") unless banner.suggestion_link_text.present?
+          add_error(banner, "is missing a survey_url") unless banner.survey_url.present?
+          add_error(banner, "is missing any page_paths") unless banner.page_paths.present?
+
+          (banner.page_paths || []).each do |path|
+            add_error(banner, "page_path #{path} should start with a /") unless path.start_with?("/")
+          end
+
+          add_error(banner, "expired over a week ago") unless (banner.end_date + 1.week) >= Time.now
```

Comment:
It was based the conversation I heard in your standup @hannako about whether a banner should be extended and the department being slow to reply. I didn't want people to be blocked by an unrelated banner.

Reply:
Not sure if an expired banner should block adding another one (only if they are on the same page) since there is a lot of involvement from other people from other departments, organisations. 
Monitoring the status of banners it's a task in itself and maybe it should be done outside these validations. Like having a slack bot that notifies the team what is live and when it should be removed?

Reply:
Yeah it's a good point. Maybe we should remove this check from the validation altogether.

It's nice that the validation helps to keep the config file neat and tidy, but being forced to sort out an unrelated banner to get an urgent one up is maybe more of a pain than having a config file with expired banners.

Reply:
(and since it's a warning, I've remove the 1 week grace period)

Reply:
Nice, sounds good 👍